### PR TITLE
Add documentation for `constant_keyword` field type

### LINKED ISSUE #7164: [DOC] add constant keyword field
**What do you want to do?**
 
- [ ] Request a change to existing documentation
- [x] Add new documentation
- [ ] Report a technical problem with the documentation
- [ ] Other 

Add new constant keyword index field type to docs
This field type was added in OS 2.14.0: opensearch-project/OpenSearch#12285

### PR DESCRIPTION
**Pull request by msfroh**
### Description
Adds documentation for the `constant_keyword` field type added in https://github.com/opensearch-project/OpenSearch/pull/12285.

### Issues Resolved
Closes [ #7164](https://github.com/opensearch-project/documentation-website/issues/7164)

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _field-types/supported-field-types/constant-keyword.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+---
+layout: default
+title: Constant Keyword
+nav_order: 46
```
Comment (msfroh) [inline _field-types/supported-field-types/constant-keyword.md:4]: Oh -- I probably need to change the `nav_order`.  This was copied from `keyword.md`.

File: _field-types/supported-field-types/constant-keyword.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+title: Constant keyword
+nav_order: 71
+has_children: false
+parent: String field types
+grand_parent: Supported field types
+---
+
+# Constant keyword field type
+
+A constant keyword field uses the same value for documents in the index.
```
Comment (natebower) [inline _field-types/supported-field-types/constant-keyword.md:12]: "documents in the index" => "all documents in an index"?

File: _field-types/supported-field-types/constant-keyword.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+has_children: false
+parent: String field types
+grand_parent: Supported field types
+---
+
+# Constant keyword field type
+
+A constant keyword field uses the same value for documents in the index. 
+
+When a search request spans multiple indexes, you can filter on a constant keyword field to match documents from indexes with the given constant value but not indexes with a different value.
```
Comment (natebower) [inline _field-types/supported-field-types/constant-keyword.md:14]:
```suggestion
When a search request spans multiple indexes, you can filter on a constant keyword field to match documents from indexes with the given constant value but not from indexes with a different value.
```

File: _field-types/supported-field-types/constant-keyword.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+## Example
+
+The following query creates a mapping with a constant keyword field.
```
Comment (natebower) [inline _field-types/supported-field-types/constant-keyword.md:18]:
```suggestion
The following query creates a mapping with a constant keyword field:
```

File: _field-types/supported-field-types/constant-keyword.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+```
+{% include copy-curl.html %}
+
+## Parameters
+
+The following table lists the parameters accepted by constant keyword field types. All values are required.
+
+Parameter | Description 
+:--- | :--- 
+`value` | The string value for the field for all documents in the index.
```
Comment (natebower) [inline _field-types/supported-field-types/constant-keyword.md:40]: "string value for the field" => "field string value"?

File: _field-types/supported-field-types/string.md [hunk 6]
```diff
@@ -20,3 +20,4 @@ Field data type | Description
 [`text`]({{site.url}}{{site.baseurl}}/opensearch/supported-field-types/text/) | A string that is analyzed. Useful for full-text search.
 [`match_only_text`]({{site.url}}{{site.baseurl}}/field-types/supported-field-types/match-only-text/) | A space-optimized version of a `text` field.
 [`token_count`]({{site.url}}{{site.baseurl}}/opensearch/supported-field-types/token-count/)  | Counts the number of tokens in a string. 
+[`constant_keyword`]({{site.url}}{{site.baseurl}}/opensearch/supported-field-types/constant-keyword/)  | Similar to `keyword`, but uses a single value for all documents.
```
Comment (natebower) [inline _field-types/supported-field-types/string.md:23]:
```suggestion
[`constant_keyword`]({{site.url}}{{site.baseurl}}/opensearch/supported-field-types/constant-keyword/)  | Similar to `keyword` but uses a single value for all documents.
```

**Review by natebower — COMMENTED** [review]
@kolchfa-aws @msfroh Please see my comments and changes and let me know if you have any questions. Thanks!

File: _field-types/supported-field-types/string.md [hunk 7]
```diff
@@ -20,3 +20,4 @@ Field data type | Description
 [`text`]({{site.url}}{{site.baseurl}}/opensearch/supported-field-types/text/) | A string that is analyzed. Useful for full-text search.
 [`match_only_text`]({{site.url}}{{site.baseurl}}/field-types/supported-field-types/match-only-text/) | A space-optimized version of a `text` field.
 [`token_count`]({{site.url}}{{site.baseurl}}/opensearch/supported-field-types/token-count/)  | Counts the number of tokens in a string. 
+[`constant_keyword`]({{site.url}}{{site.baseurl}}/opensearch/supported-field-types/constant-keyword/)  | Similar to `keyword` but uses a single value for all documents.
```
Comment (kolchfa-aws) [inline _field-types/supported-field-types/string.md:23]:
```suggestion
[`constant_keyword`]({{site.url}}{{site.baseurl}}field-types/supported-field-types/constant-keyword/)  | Similar to `keyword` but uses a single value for all documents.
```

**Review by kolchfa-aws — APPROVED** [review]
LGTM. @msfroh To what version(s) is this applicable?

File: _field-types/supported-field-types/constant-keyword.md [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
```
Comment (kolchfa-aws) [inline _field-types/supported-field-types/constant-keyword.md:40]:
```suggestion
`value` | The string field value for all documents in the index.
```

File: _field-types/supported-field-types/constant-keyword.md [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+The following query creates a mapping with a constant keyword field:
+
+```json
+PUT romcom_movies
+{
+  "mappings" : {
+    "properties" : {
+      "genre" : {
+        "value" : "Romantic comedy"
+      }
```
Comment (mrpi) [inline _field-types/supported-field-types/constant-keyword.md:27]: This example does not work. The "type" field is missing. It must look something like this:

` ...
     "genre" : {
        "type": "constant_keyword",
        "value" : "Romantic comedy"
      }
     ...
`

**Review by mrpi — COMMENTED** [review]
The provided example is being rejected by opensearch 2.14.0

**Pull request merged**